batch/batches/CaptureThumb/KAsyncCaptureThumb.class.php

```diff
@@ -211,12 +170,246 @@ private function getMediaInfoData($partnerId, $srcAssetId)
+				$thumbParamsOutput->cropWidth, $thumbParamsOutput->cropHeight, $thumbParamsOutput->stripProfiles);
+			// 5 - always force setting the given dimensions
+			$shouldResizeByPackager = KThumbnailCapture::shouldResizeByPackager($params, 5, array($thumbParamsOutput->width, $thumbParamsOutput->height));
+			list($picWidth, $picHeight) = $shouldResizeByPackager ? array($thumbParamsOutput->width, $thumbParamsOutput->height) : array(null, null);
+
+			while($count--)
+			{
+				$capturePath = $generalCapturePath . '_sec_' . $captureVidSec . '_';
+				if($packagerRetries)
+				{
+					$thumbCaptureByPackager = false;
+					$success = KThumbnailCapture::captureLocalThumbForBifUsingPackager($mediaFile, $capturePath, $captureVidSec, $picWidth, $picHeight);
+					$packagerResizeFullPath = $capturePath . self::TEMP_FILE_POSTFIX;
+					KalturaLog::debug("Packager capture is [$success] with dimension [$picWidth,$picHeight] and packagerResize [$shouldResizeByPackager] in path [$packagerResizeFullPath]");
+					if(!$success)
+					{
+						$packagerRetries--;
+						$thumbCaptureByPackager = $success;
+						$this->captureThumbUsingFfmpeg($mediaFile, $packagerResizeFullPath, $thumbParamsOutput, $mediaInfoVidDur, $mediaInfoDar, $mediaInfoScanType, $data, $mediaInfoWidth, $mediaInfoHeight, $captureVidSec, $job);
+					}
+
+					if ($thumbCaptureByPackager && $shouldResizeByPackager)
+					{
+						KalturaLog::debug("Image was resize in the packager -  setting path [$packagerResizeFullPath]");
+					}
+					else //need to crop the image
+					{
+						$this->cropFrame($packagerResizeFullPath, $data, $mediaFile, $thumbParamsOutput, $job);
+					}
+					$capturePath = $packagerResizeFullPath;
+				}
+				else
+				{
+					$this->captureThumbUsingFfmpeg($mediaFile, $capturePath, $thumbParamsOutput, $mediaInfoVidDur, $mediaInfoDar, $mediaInfoScanType, $data, $mediaInfoWidth, $mediaInfoHeight, $captureVidSec, $job);
+				}
+				$captureVidSec += $bifInterval;
+				$images[] = $capturePath;
+			}
+
+			$finalBifPath = $generalCapturePath . '.bif';
+			KalturaLog::debug("create bif file in path - [$finalBifPath]");
+
+			$bifCreator = new kBifCreator($images, $finalBifPath, $bifInterval);
+			$bifCreator->createBif();
+
+			$data->thumbPath = $finalBifPath;
+			$job = $this->moveFile($job, $data);
+
+			$files = scandir($rootPath);
+			$files = array_diff($files, array('.', '..'));
+			if($this->checkFileExists($job->data->thumbPath))
+			{
+				$updateData = new KalturaCaptureThumbJobData();
+				$updateData->thumbPath = $data->thumbPath;
+				$this->clearDir($rootPath, $files);
+				return $this->closeJob($job, null, null, null, KalturaBatchJobStatus::FINISHED, $updateData);
+			}
+
+			$this->clearDir($rootPath, $files);
+			return $this->closeJob($job, KalturaBatchJobErrorTypes::APP, KalturaBatchJobAppErrors::NFS_FILE_DOESNT_EXIST, 'File not moved correctly', KalturaBatchJobStatus::FAILED, $data);
+		}
+	}
+
+	protected function clearDir($path, $files)
```

Comment:
Move to class kFileUtils - removeFiles($dirPath , $filesList = null)
if fileList is empty , delete all files in dir